torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -500,6 +500,13 @@ class CudaKernelGenerator : private kir::IrVisitor {
     }
   }
 
+  void visit(const kir::TransposeOp* node) final {
+    TORCH_INTERNAL_ASSERT(node->out()->isA<kir::TensorIndex>());
+
+    indent() << gen(node->out()) << "\n";
```

Comment:
what exactly is the intended code generation?

Reply:
It just transposes a tensor in an out-of-place fashion.

Currently, it just copies from the input tensor to the output tensor with the specified axis permutation. Since performance optimization like avoiding uncoalesced memory accesses is possible, in the future, this `visit` function should call optimized implementations when applicable.

Reply:
I may be missing something obvious, but were does the axis permutation happen? 

Here it looks we just generate a copy between two tensors, am I wrong?

Reply:
Indexing takes care of the source and destination of each assignment.

Reply:
that makes sense - in that case, it doesn't seem we need a transpose node at all, right?